core/header/service.go

```diff
@@ -33,6 +33,10 @@ func (i *Info) Bytes() ([]byte, error) {
 	binary.LittleEndian.PutUint64(heightBytes, uint64(i.Height))
 	buf = append(buf, heightBytes...)
 
+	// TODO; permit empty hash OK for genesis block?
+	if len(i.Hash) == 0 {
```

Comment:
in baseapp empty hash is populated with hash("") @facundomedica would need to confirm this. If so then id propose we do the same

Reply:
I wasn't able to find where this happens, I'll add an issue to investigate it.